driver/src/main/java/org/neo4j/driver/internal/logging/DelegateLogger.java

```diff
@@ -0,0 +1,425 @@
+    public void trace( String msg )
+    {
+        log.trace( msg );
+    }
+
+    @Override
+    public void trace( String format, Object arg )
+    {
+        log.trace( toDriverLoggerFormat( format ), arg );
+    }
+
+    @Override
+    public void trace( String format, Object argA, Object argB )
+    {
+        log.trace( toDriverLoggerFormat( format ), argA, argB );
+    }
+
+    @Override
+    public void trace( String format, Object... arguments )
+    {
+        log.trace( toDriverLoggerFormat( format ), arguments );
+    }
+
+    @Override
+    public void trace( String msg, Throwable t )
+    {
+        log.trace( "%s%n%s", msg, t );
+    }
+
+    @Override
+    public void trace( Throwable t )
+    {
+        trace( t.getMessage(), t );
+    }
+
+    @Override
+    public boolean isDebugEnabled()
+    {
+        return log.isDebugEnabled();
+    }
+
+    @Override
+    public void debug( String msg )
+    {
+        log.debug( msg );
+    }
+
+    @Override
+    public void debug( String format, Object arg )
+    {
+        log.debug( toDriverLoggerFormat( format ), arg );
+    }
+
+    @Override
+    public void debug( String format, Object argA, Object argB )
+    {
+        log.debug( toDriverLoggerFormat( format ), argA, argB );
+    }
+
+    @Override
+    public void debug( String format, Object... arguments )
+    {
+        log.debug( toDriverLoggerFormat( format ), arguments );
+    }
+
+    @Override
+    public void debug( String msg, Throwable t )
+    {
+        log.debug( "%s%n%s", msg, t );
+    }
+
+    @Override
+    public void debug( Throwable t )
+    {
+        debug( t.getMessage(), t );
+    }
+
+    @Override
+    public boolean isInfoEnabled()
+    {
+        return true;
+    }
+
+    @Override
+    public void info( String msg )
+    {
+        log.info( msg );
+    }
+
+    @Override
+    public void info( String format, Object arg )
+    {
+        log.info( toDriverLoggerFormat( format ), arg );
+    }
+
+    @Override
+    public void info( String format, Object argA, Object argB )
+    {
+        log.info( toDriverLoggerFormat( format ), argA, argB );
+    }
+
+    @Override
+    public void info( String format, Object... arguments )
+    {
+        log.info( toDriverLoggerFormat( format ), arguments );
+    }
+
+    @Override
+    public void info( String msg, Throwable t )
+    {
+        log.info( "%s%n%s", msg, t );
+    }
+
+    @Override
+    public void info( Throwable t )
+    {
+        info( t.getMessage(), t );
+    }
+
+    @Override
+    public boolean isWarnEnabled()
+    {
+        return true;
+    }
+
+    @Override
+    public void warn( String msg )
+    {
+        log.warn( msg );
+    }
+
+    @Override
+    public void warn( String format, Object arg )
+    {
+        log.warn( toDriverLoggerFormat( format ), arg );
+    }
+
+    @Override
+    public void warn( String format, Object... arguments )
+    {
+        log.warn( toDriverLoggerFormat( format ), arguments );
```

Comment:
This method should also treat last argument in a special way if it's `Throwable`. Similar to how `#error()` works